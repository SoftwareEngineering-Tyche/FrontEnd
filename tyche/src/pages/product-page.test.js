import { render, screen } from "@testing-library/react"
import CreateNft from "./create-nft";
import ProductPage from "./product-page";
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';

test('intial product', () => {
  const wrapper = shallow(<ProductPage />);
  expect(wrapper).toMatchSnapshot();
});

test('buy button', () => {
  render(<ProductPage />);
  expect(screen.getByRole('button', { name: /خرید آنی/i })).toBeEnabled();
});

test('offer button', () => {
  render(<ProductPage />);
  expect(screen.getByRole('button', { name: /پیشنهاد قیمت/i })).toBeEnabled();
});

test('product name', () => {
  render(<ProductPage />);
  const linkElement = screen.getByText(/نام اثر/i);
  expect(linkElement).toBeInTheDocument();
});

test('product like counter', () => {
  render(<ProductPage />);
  const linkElement = screen.getByText(/نفر این اثر را پسندیده‌اند/i);
  expect(linkElement).toBeInTheDocument();
});
