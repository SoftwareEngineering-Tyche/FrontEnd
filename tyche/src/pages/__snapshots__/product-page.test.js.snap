// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`intial product 1`] = `ShallowWrapper {}`;
